homeassistant/components/sensor/qq.py

```diff
@@ -0,0 +1,72 @@
+from qqbot import _bot as bot
+from homeassistant.helpers.entity import Entity
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+import threading
+import os
+
+DOMAIN = 'qq'
+# 配置文件的样式
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required('qq'): cv.string,
+            }),
+    },
+    extra=vol.ALLOW_EXTRA)
+
+REQUIREMENTS = ['qqbot==2.3.7']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Demo sensors."""
+    o = DemoSensor()
+    add_devices([o])
+    thread1 = QQ(config['qq'])
+    thread1.start()
+
+
+class DemoSensor(Entity):
+    """Representation of a Demo sensor."""
+    def __init__(self):
+        """Initialize the sensor."""
+        self._state = '未知'
+        self._name = DOMAIN
+        self._mutex = threading.Lock()
+
+    @property
+    def should_poll(self):
+        """No polling needed for a demo sensor."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._name
+
+    def update(self):
+        path = os.path.expanduser('~') + '/.homeassistant'
```

Comment:
Why do you need to read it from a file?
Can you not get the state directly from the qqbot library?